ci/docker/Dockerfile.build.centos7

```diff
@@ -69,8 +70,8 @@ RUN yum -y check-update || true && \
         libzstd-devel && \
     yum clean all
 
-# Make GCC7, Python 3.5 and Maven 3.3 Software Collections available by default
-# during build and runtime of this container
+# Make Python 3.6 and Maven 3.3 Software Collections available by default during
+# the following build steps in this Dockerfile
 SHELL [ "/usr/bin/scl", "enable", "devtoolset-7", "rh-python36", "rh-maven35" ]
```

Comment:
I assume the goal here is to test gcc-7 in CI and gcc-9 in CD? Shall we not just entirely go to gcc-9?

Reply:
Yes, the goal is to keep ensuring that MXNet can be built with GCC 7. Switching entirely to GCC 9 may not gain much in developer productivity, as GCC 7 already has almost complete C++17 support. We could opt to adopt GCC 10 and use C++20 features, though there would be some other blockers.

The other issue is that CUDA 10 only support gcc <= 8, so for consistency of  CUDA 10 and 11 builds I'll switch this PR to use GCC 8 on CD for now.